### ISSUE TITLE
How to build and install this extension?

### ISSUE THREAD

**Issue by poke** [aspnet/LibraryManager#36]
Hey, as I’m interested in this extension and since there was no current preview VSIX that I could install, I was trying to build this myself. Building the project seemed to work correctly and I also got a VSIX out of it. However, when trying to install the VSIX, I get a message that the extension cannot be installed with my currently installed products (`NoApplicableSKUsException`). The install log mentions the following (unfortunately localized in German):

```
-------------------------------------------
Microsoft VSIX Installer
-------------------------------------------
Initializing Install...
Erweiterungsdetails...
    Identifier         : Microsoft.Web.LibraryManager
    Name               : Microsoft Library Manager
    Author             : Microsoft
    Version            : 1.0.13.1887
    Description        : Install client-side libraries easily to any web project
    Locale             : en-US
    MoreInfoURL        : https://github.com/aspnet/LibraryManager
    InstalledByMSI     : False
    SupportedFrameworkVersionRange : [4.5,)

    SignatureState     : Unsigned
    Unterstützte Produkte :
        Microsoft.VisualStudio.Community
            Version : [15.6,16.0)

    Verweise           :
        -------------------------------------------------------
        Identifier   : Microsoft.VisualStudio.MPF.15.0
        Name         : Visual Studio MPF 15.0
        Version      : [15.0]
        MoreInfoURL  :
        Nested       : No

    Voraussetzungen    :
        -------------------------------------------------------
        Identifier   : Microsoft.VisualStudio.Component.CoreEditor
        Name         : Visual Studio core editor
        Version      : [15.0,16.0)

        -------------------------------------------------------
        Identifier   : Microsoft.VisualStudio.ComponentGroup.WebToolsExtensions
        Name         : ASP.NET and web development
        Version      : [15.0,16.0)

Signaturdetails...
    Extension is not signed.

Er wird nach passenden Produkten gesucht...
Installiertes Produkt gefunden - Globaler Pfad
Installiertes Produkt gefunden - Visual Studio Enterprise 2017
VSIXInstaller.NoApplicableSKUsException: Diese Erweiterung kann auf den derzeit installierten Produkten nicht installiert werden.
   bei VSIXInstaller.App.GetInstallableData(String vsixPath, Boolean isRepairSupported, IEnumerable`1& skuData)
   bei VSIXInstaller.App.Initialize(Boolean isRepairSupported)
   bei VSIXInstaller.App.Initialize()
   bei System.Threading.Tasks.Task`1.InnerInvoke()
   bei System.Threading.Tasks.Task.Execute()
```

As the logs note, I have VS Enterprise 2017 installed, version 15.6.4. The `vsixmanifest` specifies `Microsoft.VisualStudio.Community` as the installation target with the version range `[15.6,16.0)`. So the installation should work fine if I understand this correctly.

Can you give me some information on what I need to do to get this up and running?

**Comment by nickalbrecht** [comment]
Any guidance on trying this out? Been watching this repo for a while now and would love to try it out.

**Comment by jimmylewis** [comment]
I'm investigating why this is happening; we had to make some changes in order to get this to ship as an in-box component in VS which seem to be affecting it.  

As a workaround if you clone the repo and F5, it will install to the Exp instance and you can play with it there.

(Caveat: there are still a number of bugs that we've been tracking internally, but feel free to open issues here on anything you run into.)

**Comment by watrush** [comment]
Same problem.
28.03.2018 10:05:55 - Microsoft VSIX Installer
28.03.2018 10:05:55 - -------------------------------------------
28.03.2018 10:05:55 - vsixinstaller.exe version:
28.03.2018 10:05:55 - 15.6.152+ge95e4dd1ce
28.03.2018 10:05:55 - -------------------------------------------
28.03.2018 10:05:55 - Command line parameters:
28.03.2018 10:05:55 - C:\Program Files (x86)\Microsoft Visual Studio\Installer\resources\app\ServiceHub\Services\Microsoft.VisualStudio.Setup.Service\VSIXInstaller.exe,C:\Users\kucherov\source\repos\LibraryManager\src\LibraryManager.Vsix\bin\Release\Microsoft.Web.LibraryManager.vsix
28.03.2018 10:05:55 - -------------------------------------------
28.03.2018 10:05:55 - Microsoft VSIX Installer
28.03.2018 10:05:55 - -------------------------------------------
28.03.2018 10:06:05 - Initializing Install...
28.03.2018 10:06:05 - Подробности расширения...
28.03.2018 10:06:05 - 	Identifier         : Microsoft.Web.LibraryManager
28.03.2018 10:06:05 - 	Name               : Microsoft Library Manager
28.03.2018 10:06:05 - 	Author             : Microsoft
28.03.2018 10:06:05 - 	Version            : 1.0.16.16678
28.03.2018 10:06:05 - 	Description        : Install client-side libraries easily to any web project
28.03.2018 10:06:05 - 	Locale             : en-US
28.03.2018 10:06:05 - 	MoreInfoURL        : https://github.com/aspnet/LibraryManager
28.03.2018 10:06:05 - 	InstalledByMSI     : False
28.03.2018 10:06:05 - 	SupportedFrameworkVersionRange : [4.5,)
28.03.2018 10:06:05 - 
28.03.2018 10:06:05 - 	SignatureState     : Unsigned
28.03.2018 10:06:05 - 	Поддерживаемые продукты : 
28.03.2018 10:06:05 - 		Microsoft.VisualStudio.Community
28.03.2018 10:06:05 - 			Version : [15.6,16.0)
28.03.2018 10:06:05 - 
28.03.2018 10:06:05 - 	Ссылки             : 
28.03.2018 10:06:05 - 		-------------------------------------------------------
28.03.2018 10:06:05 - 		Identifier   : Microsoft.VisualStudio.MPF.15.0
28.03.2018 10:06:05 - 		Name         : Visual Studio MPF 15.0
28.03.2018 10:06:05 - 		Version      : [15.0]
28.03.2018 10:06:05 - 		MoreInfoURL  : 
28.03.2018 10:06:05 - 		Nested       : No
28.03.2018 10:06:05 - 
28.03.2018 10:06:05 - 	Предварительные требования : 
28.03.2018 10:06:05 - 		-------------------------------------------------------
28.03.2018 10:06:05 - 		Identifier   : Microsoft.VisualStudio.Component.CoreEditor
28.03.2018 10:06:05 - 		Name         : Visual Studio core editor
28.03.2018 10:06:05 - 		Version      : [15.0,16.0)
28.03.2018 10:06:05 - 
28.03.2018 10:06:05 - 		-------------------------------------------------------
28.03.2018 10:06:05 - 		Identifier   : Microsoft.VisualStudio.ComponentGroup.WebToolsExtensions
28.03.2018 10:06:05 - 		Name         : ASP.NET and web development
28.03.2018 10:06:05 - 		Version      : [15.0,16.0)
28.03.2018 10:06:05 - 
28.03.2018 10:06:05 - Состав подписи...
28.03.2018 10:06:05 - 	Extension is not signed.
28.03.2018 10:06:05 - 
28.03.2018 10:06:05 - Поиск применимых продуктов...
28.03.2018 10:06:05 - Найден установленный продукт - Глобальное расположение
28.03.2018 10:06:05 - Найден установленный продукт - ssms
28.03.2018 10:06:05 - Найден установленный продукт - ssms
28.03.2018 10:06:05 - Найден установленный продукт - Visual Studio Enterprise 2017
28.03.2018 10:06:05 - VSIXInstaller.NoApplicableSKUsException: Это расширение не может быть установлено ни для одного из установленных продуктов.
   в VSIXInstaller.App.GetInstallableData(String vsixPath, Boolean isRepairSupported, IEnumerable`1& skuData)
   в VSIXInstaller.App.Initialize(Boolean isRepairSupported)
   в VSIXInstaller.App.Initialize()
   в System.Threading.Tasks.Task`1.InnerInvoke()
   в System.Threading.Tasks.Task.Execute()

**Comment by poke** [comment]
> As a workaround if you clone the repo and F5, it will install to the Exp instance and you can play with it there.

I did try that but for some reason the launched instance was completely unresponsive for me. But that was probably an issue with my computer. I will give it another try later.

**Comment by nickalbrecht** [comment]
@poke The Experimental Instance is slow to come up, but it does work. I decided to install the VSIX tooling to give this a shot.

**Comment by jimmylewis** [comment]
This will get addressed by #44

**Comment by poke** [comment]
Just to confirm: I was able to compile and install the VSIX now. Thanks! :)
